README.md

```diff
@@ -1,105 +1,366 @@
+</div>
 ```
-<fa-icon [icon]="faUser"></fa-icon>
+
+`src/app/app.component.ts`
+
+```typescript
+import { Component } from '@angular/core';
+import { faCoffee } from '@fortawesome/free-solid-svg-icons';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+  title = 'app';
+  faCoffee = faCoffee;
+}
 ```
 
-Run `yarn start` or `npm run start` to see more example uses.
+`src/app/app.module.ts`
 
-## Advanced Usage
+1. Import `{ FontAwesomeModule } from '@fortawesome/angular-fontawesome'`
+1. Add `FontAwesomeModule` to `imports`
 
-*With Mask and Transform*
+```typescript
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+
+import { AppComponent } from './app.component';
+import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
+
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    BrowserModule,
+    FontAwesomeModule
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
 ```
-<fa-icon [icon]="faCircle" transform="shrink-9 right-4" [mask]="faSquare"></fa-icon>
+
+### Using the Font Awesome Library
+
+The library provides convenient usage in your templates but you have to manage
+the icons separate from your components. This means that if someone
+accidentally removes the icon your component which uses it could break.
+
+`src/app/app.component.html`
+
+```html
+<div style="text-align:center">
+  <!-- simple name only that assumes the 'fas' prefix -->
+  <fa-icon icon="coffee"></fa-icon>
+  <!-- ['fas', 'coffee'] is an array that indicates the [prefix, iconName] -->
+  <fa-icon [icon]="['fas', 'coffee']"></fa-icon>
+</div>
 ```
 
-*Spin animation with click toggle*
+`src/app/app.module.ts`
+
+1. Import `{ FontAwesomeModule } from '@fortawesome/angular-fontawesome'`
+1. Add `FontAwesomeModule` to `imports`
+1. Import `{ library } from '@fortawesome/fontawesome-svg-core'`
+1. Import an icon like `{ faCoffee } from '@fortawesome/free-solid-svg-icons'`
+1. Add to the library with `library.add(faCoffee)`
+
+```typescript
+import { BrowserModule } from '@angular/platform-browser';
+import { NgModule } from '@angular/core';
+
+import { AppComponent } from './app.component';
+import { FontAwesomeModule } from '@fortawesome/angular-fontawesome';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faCoffee } from '@fortawesome/free-solid-svg-icons';
+
+// Add an icon to the library for convenient access in other components
+library.add(faCoffee);
+
+@NgModule({
+  declarations: [
+    AppComponent
+  ],
+  imports: [
+    BrowserModule,
+    FontAwesomeModule
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
 ```
-<fa-icon [icon]="faSync" [spin]="isSyncAnimated" (click)="isSyncAnimated=!isSyncAnimated"></fa-icon>
+
+You can also import entire styles (but be careful!).
```

Comment:
Maybe clarify why - it will significantly increase bundle size.

Reply:
I'll add some elaboration on that while I'm already editing this thing.